### ISSUE TITLE
non linear Colorbar issue with dfs.plot

### ISSUE THREAD

**Issue by jouArc** [DHI/mikeio#193]
**Describe the bug**
I have an issue with non linear colorbar when I try to plot 3D dfsu with dfs.plot, the smallest values are all of the same color instead of the ones in the indicated colobar. 

```
cmap = mpl.colors.ListedColormap(['cyan','green','orange','red'])
bounds = [0, 100, 500, 1000, 2000]

ax=dfs.plot(maxvarALL_f, plot_type='contourf', elements=dfs.top_elements, show_mesh=False,
            cmap = cmap,
            vmin=0,
            vmax=2000,
            levels=bounds)
```
I have this :
![image](https://user-images.githubusercontent.com/81360896/112465235-c8338e00-8d64-11eb-9f78-c2157d1d1612.png)

Instead of this colorbar :
![image](https://user-images.githubusercontent.com/81360896/112464877-53f8ea80-8d64-11eb-9a03-fd0047852d04.png)

Many thanks for the help, I'm new to python :)

**System information:**
 - Python version : 3.7 
 - MIKE version : 2021

**Comment by jsmariegaard** [comment]
Hi @jouArc 
It took me a while to find time to look into this - but now I did :-) 

It seems to be a matplotlib issue - take a look at this similar problem: https://stackoverflow.com/questions/52842553/matplotlib-listedcolormap-not-mapping-colors 

Apparantly it happens because you have non-equidistant bounds. Try to look at the norm argument to the above answer. Do you think this would work for you?

**Comment by jouArc** [comment]
Hi,
thank for your answer! :)
I already looked at the norm argument, I have no problem doing non linear colorbar with matplotlib only using the norm argument

For example here : 

```
fig, ax = plt.subplots(figsize=(6, 1))
fig.subplots_adjust(bottom=0.5)

cmap = mpl.colors.ListedColormap(['cyan','green','orange','red'])
bounds = [0, 100, 500, 1000, 2000]
norm = mpl.colors.BoundaryNorm(bounds, cmap.N)

plt.colorbar(
    mpl.cm.ScalarMappable(cmap=cmap, norm=norm),
    cax=ax,
    boundaries=bounds,  # Adding values for extensions.
    extend='neither',
    ticks=bounds,
    spacing='uniform',
    orientation='horizontal',
    label='Conc. E.Coli',
)
```
But I do not know how to use the norm argument with the **dfs.plot** function in the mikeio package. 

norm argument is not a entry parameter and I do not know how to change the dfs.plot function to use it ? I tried different things but nothing worked. 
Could it be an enhancement or is there something else to do ?
Thanks again!!

**Comment by jsmariegaard** [comment]
Closed by #195
